Tests/OrderedCollectionsTests/OrderedSet/OrderedSet Diffing Tests.swift

```diff
@@ -0,0 +1,142 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import OrderedCollections
+import CollectionsTestSupport
+
+class MeasuringHashable: Hashable {
+  static var equalityChecks = 0
+  static func == (lhs: MeasuringHashable, rhs: MeasuringHashable) -> Bool {
+    MeasuringHashable.equalityChecks += 1
+    return lhs._inner == rhs._inner
+  }
+
+  static var hashChecks = 0
+  func hash(into hasher: inout Hasher) {
+    MeasuringHashable.hashChecks += 1
+    _inner.hash(into: &hasher)
+  }
+
+  let _inner: AnyHashable
+
+  init<T: Hashable>(_ wrapped: T) {
+    _inner = AnyHashable(wrapped)
+  }
+}
+
+class OrderedSetDiffingTests: CollectionTestCase {
+
+  func _validatePerformance<T: Hashable>(from a: OrderedSet<T>, to b: OrderedSet<T>) {
+    MeasuringHashable.equalityChecks = 0
+    MeasuringHashable.hashChecks = 0
+
+    let _ = OrderedSet(a.map({MeasuringHashable($0)})).difference(from: OrderedSet(b.map({MeasuringHashable($0)})))
+    let n = a.count + b.count
+
+    /* Expect linear performance, which we can fence in testing as
+     * "less than nlogn" since diffing generally tends to be at least n**2
+     */
+    expectLessThan(MeasuringHashable.equalityChecks, n * n.bitWidth)
+    expectLessThan(MeasuringHashable.hashChecks, n * n.bitWidth)
```

Comment:
💯💯💯

Reply:
Why doesn't the stdlib have a logarithm function anyway?

Reply:
BinaryInteger does have a customization point for it, it just isn't public yet -- it's currently called `_binaryLogarithm()`. (By default it is defined in terms of `leadingZeroBitCount`.)